CAN_STM32: Build error with CONFIG_CAN_AUTO_BUS_OFF_RECOVERY=n

Build error:
`zephyr/drivers/can/can_stm32.c:596:15: error: invalid operands to binary != (have 'k_timeout_t' {aka 'struct <anonymous>'} and 'k_timeout_t' {aka 'struct <anonymous>'})`

Steps to reproduce:
1. In **samples/drivers/can/prj.conf** add `CONFIG_CAN_AUTO_BUS_OFF_RECOVERY=n`
2. run `west build -b nucleo_f446re samples/drivers/can`

Test Environment:
  Linux (Debian 'sid') using `arm-none-eabi-gcc (15:8-2019-q3-1+b1) 8.3.1 20190703 (release) [gcc-8-branch revision 273027]`
  ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
  CROSS_COMPILE=/usr/bin/arm-none-eabi-

Versions affected: 
  Latest (ce864f6e)

--- a/drivers/can/can_mcux_flexcan.c
+++ b/drivers/can/can_mcux_flexcan.c
@@ -469,9 +469,9 @@ int mcux_flexcan_recover(const struct device *dev, k_timeout_t timeout)
 	start_time = k_uptime_ticks();
 	config->base->CTRL1 &= ~CAN_CTRL1_BOFFREC_MASK;
 
-	if (timeout != K_NO_WAIT) {
+	if (!K_TIMEOUT_EQ(timeout, K_NO_WAIT)) {
 		while (mcux_flexcan_get_state(dev, NULL) == CAN_BUS_OFF) {
-			if (timeout != K_FOREVER &&
+			if (!K_TIMEOUT_EQ(timeout, K_FOREVER) &&
 			    k_uptime_ticks() - start_time >= timeout.ticks) {
 				ret = CAN_TIMEOUT;
 			}

--- a/drivers/can/can_stm32.c
+++ b/drivers/can/can_stm32.c
@@ -593,7 +593,7 @@ int can_stm32_recover(const struct device *dev, k_timeout_t timeout)
 	start_time = k_uptime_ticks();
 
 	while (can->ESR & CAN_ESR_BOFF) {
-		if (timeout != K_FOREVER &&
+		if (!K_TIMEOUT_EQ(timeout, K_FOREVER) &&
 		    k_uptime_ticks() - start_time >= timeout.ticks) {
 			goto done;
 		}

